src/dmd/expressionsem.d

```diff
@@ -6772,7 +6765,6 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
              */
             assert(exp.e1.op == TOK.star);
             PtrExp pe = cast(PtrExp)exp.e1;
-            assert(pe.e1.op == TOK.question);
             CondExp ce = cast(CondExp)pe.e1;
```

Comment:
Use isCondExp

Reply:
Also, why remove this but leave the assert prior?

Reply:
it's verified, see upper, by the variable wasCond.